src/core.rs

```diff
@@ -127,7 +127,18 @@ pub async fn plan_add(
 
 /// Read a Plan Dag from the DB
 pub async fn plan_read(tx: &mut Transaction<'_, Postgres>, id: DatasetId) -> poem::Result<Plan> {
-    Plan::from_dataset_id(tx, id).await.map_err(to_poem_error)
+    Plan::from_db(tx, id).await.map_err(to_poem_error)
+}
+
+/// Read a Data Product from the DB
```

Comment:
@coderabbitai Please add a unit test where this function can read a data product record from the DB, and a unit test where there data product does not exists.

Reply:
Please create a PR for this change.